Doc/library/random.rst

```diff
@@ -30,8 +30,12 @@ deterministic, it is not suitable for all purposes, and is completely unsuitable
 for cryptographic purposes.
 
 The functions supplied by this module are actually bound methods of a hidden
-instance of the :class:`random.Random` class.  You can instantiate your own
-instances of :class:`Random` to get generators that don't share state.
+instance of the :class:`random.Random` class, and thus share state – in 
+particular, calls to :meth:`random.set_seed` anywhere in application code 
+or any libraries will affect the sequence of random values produced by calls 
+to `random.*` methods in the same process, which can lead to unexpected results.
+You should consider instantiating your own instance or instances of :class:`Random` 
+to get generators that don't share state. 
```

Comment:
For some applications, the sharing of state might not matter -- perhaps provide a heuristic for where I'd want to instantiate my own `Random` class?